src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOfAnalyzer.cs

```diff
@@ -0,0 +1,184 @@
+                    charAndComparisonTypeArgumentContainsMethod == null ||
+                    charArgumentContainsMethod == null)
+                {
+                    return;
+                }
+
+                // Roslyn doesn't yet support "FindAllReferences" at a file/block level. So instead, find references to local int variables in this block.
+                context.RegisterOperationBlockStartAction(OnOperationBlockStart);
+
+                void OnOperationBlockStart(OperationBlockStartAnalysisContext context)
+                {
+                    if (!(context.OwningSymbol is IMethodSymbol method))
+                    {
+                        return;
+                    }
+
+                    PooledConcurrentSet<ISymbol> localsToBailOut = PooledConcurrentSet<ISymbol>.GetInstance();
+                    PooledConcurrentDictionary<ISymbol, IInvocationOperation> variableNameToOperationsMap = PooledConcurrentDictionary<ISymbol, IInvocationOperation>.GetInstance();
+
+                    context.RegisterOperationAction(PopulateLocalReferencesSet, OperationKind.LocalReference);
+
+                    context.RegisterOperationAction(AnalyzeInvocationOperation, OperationKind.Invocation);
+
+                    context.RegisterOperationBlockEndAction(OnOperationBlockEnd);
+
+                    // Local Functions
+                    void PopulateLocalReferencesSet(OperationAnalysisContext context)
+                    {
+                        ILocalReferenceOperation localReference = (ILocalReferenceOperation)context.Operation;
+                        if (localReference.Local.Type.SpecialType != SpecialType.System_Int32)
+                        {
+                            return;
+                        }
+
+                        var parent = localReference.Parent;
+                        if (!(parent is IBinaryOperation binaryOperation))
+                        {
+                            localsToBailOut.Add(localReference.Local);
+                        }
+                    }
+
+                    void AnalyzeInvocationOperation(OperationAnalysisContext context)
+                    {
+                        var invocationOperation = (IInvocationOperation)context.Operation;
+                        if (!IsDesiredTargetMethod(invocationOperation.TargetMethod))
+                        {
+                            return;
+                        }
+
+                        var parent = invocationOperation.Parent;
+                        if (parent is IBinaryOperation blockOperation)
+                        {
+                            var operatorKind = blockOperation.OperatorKind;
+                            if (operatorKind != BinaryOperatorKind.Equals && operatorKind != BinaryOperatorKind.GreaterThanOrEqual)
+                            {
+                                return;
+                            }
+
+                            var rightOperand = blockOperation.RightOperand;
```

Comment:
I think you should handle both cases `index op constant` and `constant op index`, i.e.:

```csharp
var otherOperand = binaryOperation.LeftOperand is invocationOperation ? binaryOperation.RightOperand : binaryOperation.LeftOperand;
```